consult.el

```diff
@@ -44,6 +44,7 @@
 
 (require 'bookmark)
 (require 'cl-lib)
+(require 'flymake)
```

Comment:
Please move the command to an extra file `consult-flymake.el` such that flymake is not always loaded. This makes sense for the flycheck users. The file `consult-flycheck.el` can still be distributed as part of the MELPA `consult` package in contrast to `consult-flycheck`.

Reply:
Why is unnecessarily loading Flymake a problem? It only activates if flymake-mode enables, and thus wouldn't impact Flycheck users.

Reply:
Generally it is better to avoid loading any unnecessary dependency. We will do the same if we add consult-org-* commands. They will go to a consult-org.el file. The dependencies of consult.el should be rather lightweight as of now.

Reply:
That makes sense. How about lazily requiring flymake, per https://github.com/minad/consult/pull/66/commits/af1401cf8cbe0edd14906d2a3dae49f8221f49c4 ?

This allows non-Flymake users to avoid having to load Flymake, but allows `consult-flymake` to be able and discoverable by all users.

Reply:
@tomfitzhenry In case you worry. This will not break lazy loading. It will just work. Everything will work if it is in an extra file. The only advantage is that flymake is not loaded for users who don't need it. Furthermore this lazy requiring breaks package-lint is therefore not a good idea.

Reply:
Ah, yes. Fair point. Autoload will continue to work.  Ok, I'll move it to a separate file.

Reply:
Done.